### PR TITLE
Fix issues with PopupHelpWidget on Linux and macOS

### LINKED ISSUE #2814: Advanced Search Issues "Search Help" won't stay visible, cannot do advanced search
## Expected Behavior
I expected to be able to do an advanced search.

## Current Behavior
Clicking on the green help icon for search help causes a popup display that immediately disappears, so it isn't useful.

## Steps to Reproduce
1. Open database and unlock database with v2.4.0 version
2. Click on "Help icon" that has a tool tip of "Search Help"

The actual "Help" menu has nothing about "Help" at all either.

## Context
Can't use advanced help, which I was looking forward to.  Non-advanced help still works as before.

## Debug Info
KeePassXC - Version 2.4.0
Revision: c51752d
Distribution: AppImage

Libraries:
- Qt 5.10.1
- libgcrypt 1.8.1

Operating system: Devuan GNU/Linux ascii
CPU architecture: x86_64
Kernel: linux 4.9.0-6-amd64 [SMP Debian 4.9.88-1+deb9u1 (2018-05-07)]

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (only unsigned sharing)
- YubiKey

### PR DESCRIPTION
**Pull request by droidmonkey**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )

* Clean up parent alignment code
* Don't hide widget if it currently has focus
* Use Qt::Tool window type on macOS as well. This prevents the popup help from hiding to the background if the main window has focus.
* Fixes #2814

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested on Windows, Linux, and macOS

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: src/gui/widgets/PopupHelpWidget.cpp [hunk 1]
```diff
@@ -65,10 +60,10 @@ void PopupHelpWidget::setPosition(Qt::Corner corner)
...
 bool PopupHelpWidget::eventFilter(QObject* obj, QEvent* event)
 {
-    if (obj == parentWidget() && event->type() == QEvent::FocusOut) {
-        hide();
-    } else if (obj == m_appWindow && (event->type() == QEvent::Move || event->type() == QEvent::Resize)) {
-        if (isVisible()) {
+    if (isVisible()) {
+        if (obj == parentWidget() && event->type() == QEvent::FocusOut && qApp->focusWindow() != windowHandle()) {
+            hide();
+        } else if (obj == m_appWindow && (event->type() == QEvent::Move || event->type() == QEvent::Resize)) {
```
Comment (phoerious) [inline src/gui/widgets/PopupHelpWidget.cpp]: Combine conditions or negate and return early.
To make the lines shorter and the intent clear, this whole thing could even be reformulated in terms of `event->type()` and everything else as secondary conditions.

Comment (droidmonkey) [reply src/gui/widgets/PopupHelpWidget.cpp]: I don't see the problem, we only want to listen to the MainWindow move/resize events in this block. It is distinct from the block above where we are listening for our parent widget's events. This is the most compact and readable the code can get IMO.

Comment (droidmonkey) [reply src/gui/widgets/PopupHelpWidget.cpp]: @phoerious any comment?

Comment (phoerious) [reply src/gui/widgets/PopupHelpWidget.cpp]: I'd see the event type as the main condition and visibility as a secondary decision criterion. But it's your call, in the end I'm fine with either.

**Pull request merged**